### ISSUE TITLE
Object literal may only specify known properties, and 'id' does not exist in type

### ISSUE THREAD

**Issue by ellipticaldoor** [prisma/prisma2#517]
I'm trying to create a record with a fixed id that is an Int but I get the following error
```
src/server/debug/seed.ts:35:9 - error TS2322: Type '{ id: number; ticks: number; }' is not assignable to type 'GameStateCreateInput'.
  Object literal may only specify known properties, and 'id' does not exist in type 'GameStateCreateInput'.

35         id: DEFAULT_GAME_STATE_ID,
```

This is how the generated type for "GameStateCreateInput" looks like
``` typescript
export declare type GameStateCreateInput = {
    ticks: number;
};
```

This is how the code looks like
``` javascript
await photon.gameStates.create({
  data: {
    id: DEFAULT_GAME_STATE_ID,
    ticks: 0,
  },
})
```

And this is the model I'm using
```
model GameState {
  id Int @id
  ticks Int
}
```

I find this behaviour strange because when the "id" is an "String" I can set the id on creation.

Is there a fix for this?

**Comment by pantharshit00** [comment]
This is a bug indeed. I can confirm this.


Right now an Integer id is unavailable in the `CreateInput`.

As a workaround, you can use string IDs for now:
```
model GameState {
  id String @id @default(cuid())
  ticks Int
}
```

```js
console.log(typeof DEFAULT_GAME_STATE_ID); // string
await photon.gameStates.create({
  data: {
    id: DEFAULT_GAME_STATE_ID,
    ticks: 0,
  },
})
```

**Comment by pantharshit00** [comment]
Posted a reproduction here: https://github.com/harshit-test-org/prisma-prisma2-517

**Comment by ellipticaldoor** [comment]
Is there any update on this one?

**Comment by janpio** [comment]
Internal Note: Spec needed, ID behaviour.

**Comment by timsuchanek** [comment]
@mavilein will leave a note, why this is not that trivial.

**Comment by mavilein** [comment]
So we have to distinguish between 2 cases here in my opinion:
1. A model with an id field of type int, where the id is auto generated/incremented. In this case we do **not** want to allow users to supply an own id. Why? Imagine a table where the last row has id 10. Now a user manually inserts a row with id 11. From then on all inserts without an explicit id would fail because the database would try to use 11 as the next id which already exists now.
2. A model with an id field of type int, where the id is not auto generated. In this case we must allow users to supply an id.

Currently all models with an id field like this `id Int @id` fall into case 1. There's no way right now to opt into case 2. There's already an [issue](https://github.com/prisma/specs/issues/176) in progress that address this problem.

**Comment by mavilein** [comment]
This was fixed by @do4gr in https://github.com/prisma/prisma-engine/issues/274

**Comment by jontroncoso** [comment]
A version of this is still happening for me.

I'm using the "GraphQL+Auth", Typescript, code-sample and building off of that. 

I noticed this error highlighted in my IDE, in `src/types/Query.ts`:
<img width="756" alt="Screen Shot 2020-01-20 at 5 31 11 PM" src="https://user-images.githubusercontent.com/300521/72767942-b677b900-3baa-11ea-9933-5a5799e63d72.png">

... I am using this in my schema.prisma:
```
  id                       Int      @id @default(autoincrement())
```

... so I decided to `npm run generate && npm run build`. On the `npm run build step`, I got:

```
src/types/Query.ts:60:13 - error TS2322: Type 'string | null | undefined' is not assignable to type 'number | null | undefined'.
  Type 'string' is not assignable to type 'number | null | undefined'.

60             id,
               ~~
```
